client/signup/steps/woocommerce-install/step-store-address/index.tsx

```diff
@@ -162,3 +203,56 @@ export default function StepStoreAddress( props: WooCommerceInstallProps ): Reac
 		/>
 	);
 }
+
+function ControlError( props: { error: string } ): ReactElement | null {
+	const { error } = props;
+	if ( error ) {
+		return <FormInputValidation isError={ true } isValid={ false } text={ error } />;
+	}
+	return null;
+}
+
+function useAddressFormValidation( siteId: number ) {
+	const { get } = useSiteSettings( siteId );
+	const { __ } = useI18n();
+
+	const [ errors, setErrors ] = useState( {} as Record< optionNameType, string > );
+
+	const validate = () => {
+		errors[ WOOCOMMERCE_STORE_ADDRESS_1 ] = ! get( WOOCOMMERCE_STORE_ADDRESS_1 )
+			? __( 'Address line is required.' )
+			: '';
+		errors[ WOOCOMMERCE_STORE_ADDRESS_2 ] = ''; // Optional field
+		errors[ WOOCOMMERCE_DEFAULT_COUNTRY ] = ! get( WOOCOMMERCE_DEFAULT_COUNTRY )
+			? __( 'Country / Region is required.' )
+			: '';
+		errors[ WOOCOMMERCE_STORE_CITY ] = ! get( WOOCOMMERCE_STORE_CITY )
+			? __( 'City is required.' )
+			: '';
+		errors[ WOOCOMMERCE_STORE_POSTCODE ] = ! get( WOOCOMMERCE_STORE_POSTCODE )
+			? __( 'Postcode is required.' )
+			: '';
+		errors[ WOOCOMMERCE_ONBOARDING_PROFILE ] = ! get( WOOCOMMERCE_ONBOARDING_PROFILE )[
+			'store_email'
+		]
+			? __( 'Email address is required.' )
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **96** times:
`translate( 'A valid email address is required' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woop-validation)